go/libraries/doltcore/rowconv/super_schema.go

```diff
@@ -111,55 +145,93 @@ func (ss *SuperSchemaGen) AddSchema(sch schema.Schema) error {
 	return nil
 }
 
+func (ssg *SuperSchemaGen) nameCols() map[TagKindPair]string {
+	colNames := map[string][]TagKindPair{}
+	for tagKind, names := range ssg.names {
+		name := fmt.Sprintf("%d_%s", tagKind.Tag, tagKind.Kind.String())
+		if names.Size() == 1 {
+			name = names.AsSlice()[0]
+		}
+		colNames[name] = append(colNames[name], tagKind)
+	}
+
+	results := map[TagKindPair]string{}
+	for name, tagKinds := range colNames {
+		if len(tagKinds) == 1 {
+			results[tagKinds[0]] = name
+			continue
+		}
+
+		for _, tagKind := range tagKinds {
+			name := fmt.Sprintf("%s_%s_%d", name, tagKind.Kind.String(), tagKind.Tag)
+			results[tagKind] = name
+		}
+	}
+
+	return results
+}
+
 // GenerateSuperSchema takes all the accumulated schemas and generates a schema which is the superset of all of
 // those schemas.
-func (ss *SuperSchemaGen) GenerateSuperSchema(additionalCols ...NameKindPair) (schema.Schema, error) {
+func (ssg *SuperSchemaGen) GenerateSuperSchema(additionalCols ...NameKindPair) (SuperSchema, error) {
+	namedCols := ssg.nameCols()
+
 	colColl, _ := schema.NewColCollection()
-	for tagKind, destTag := range ss.tagKindToDestTag {
-		colName := fmt.Sprintf("%d_%s", tagKind.Tag, tagKind.Kind.String())
+	for tagKind, colName := range namedCols {
+		destTag, ok := ssg.tagKindToDestTag[tagKind]
+
+		if !ok {
+			panic("mismatch between namedCols and tagKindToDestTag")
```

Comment:
More appropriate to return an error here?

Reply:
if we hit this, there is a bug.  Likely it occurred when adding schemas.  Want to panic, because it should never get here.